File: 1441-minimum-flips-to-make-a-or-b-equal-to-c/minimum-flips-to-make-a-or-b-equal-to-c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        x = format(a, '032b')
        y = format(b, '032b')
        z = format(c, '032b')
        print(x," + ",y," + ",z)
        count = 0
        for i in range(len(z)-1,-1,-1):
            if x[i]=='1' and y[i]=='1' and z[i]=='0':
                count+=2
            elif x[i]=='1' and y[i]=='0' and z[i]=='0':
                count+=1
            elif x[i]=='0' and y[i]=='1' and z[i]=='0':
                count+=1
            elif x[i]=='0' and y[i]=='0' and z[i]=='1':
                count+=1
        
        return count

File: 1441-minimum-flips-to-make-a-or-b-equal-to-c/minimum-flips-to-make-a-or-b-equal-to-c.cpp
class Solution {
public:
   vector<int> decToBinary(int n ){
    
   vector<int>binaryNum(32,0);
 
   
    int i = 0;
    while (n > 0) {
 
       
        binaryNum[i] = n % 2;
        n = n / 2;
        i++;
    }
    return binaryNum;
     }
    int minFlips(int a, int b, int c) {
        
        
    vector<int>binarya=decToBinary(a );
    vector<int> binaryb=decToBinary( b );
    vector<int> binaryc=decToBinary( c );
     int ans=0;
    
     
     for(int i=0;i<32;i++){
         if((binarya[i] | binaryb[i])==binaryc[i]){
             
             continue;
         }
         else if(binaryc[i]==0){
             if(binarya[i]==1 && binaryb[i]==1){
                 
                 ans=ans+2;
                //  continue;
             }
             else{
                 
                 ans++;
                 continue;
             }
         }
         else if(binaryc[i]==1) {
             
              if(binarya[i]==0 && binaryb[i]==0){
                  ans++;
              }
             
            //  continue;
         }
     }
     return ans;
       
    }
};